lightning/src/onion_message/messenger.rs

```diff
@@ -735,6 +735,8 @@ where
 		&self, contents: T, destination: Destination, reply_path: Option<BlindedPath>,
 		log_suffix: fmt::Arguments
 	) -> Result<SendSuccess, SendError> {
+		let intro_node_logger =
+			WithContext::from(&self.logger, Some(destination.first_node()), None);
```

Comment:
Hmmm... it's not clear to me we want to use `destination.first_node()`. With `DefaultMessageRouter` it will either be our peer or soon-to-be peer, which I believe is what we want. But a different `MessageRouter` may return a partial `path` back from `destination`, in which case the peer (or soon-to-be peer) in question would not be the introduction node.